byway/src/main/webapp/recommendations.js

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global google */
+/* global destination:writable, step:writable, interest:writable, region:writable, result:writable */
+
+if(document.readyState === 'loading') {
+  document.addEventListener('DOMContentLoaded', initialize);
+} else {
+  initialize();
+}
+
+let map;
+let placesService;
+
+// Measured in meters
+let RADIUS = 1000;
+let MIN_DISTANCE = 4000;
+
+let interests = ["park"];
+let destinations = new Set();
+let regions = [];
```

Comment:
May be good to comment what exactly regions holds as it is not intuitive.

Reply:
Updated